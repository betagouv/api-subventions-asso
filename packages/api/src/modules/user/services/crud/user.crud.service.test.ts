import userCrudService from "./user.crud.service";
import userRepository from "../../repositories/user.repository";
import { USER_EMAIL } from "../../../../../tests/__helpers__/userHelper";
jest.mock("../../repositories/user.repository");
const mockedUserRepository = jest.mocked(userRepository);

describe("user crud service", () => {
    describe("findByEmail", () => {
        it("should call userRepository.findByEmail", async () => {
            await userCrudService.findByEmail(USER_EMAIL);
            expect(mockedUserRepository.findByEmail).toHaveBeenCalledWith(USER_EMAIL);
        });
    });
});
