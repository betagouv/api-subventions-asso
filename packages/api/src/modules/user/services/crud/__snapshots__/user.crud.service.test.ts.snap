// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user crud service listUsers returns proper result 1`] = `
[
  {
    "_id": 1,
    "resetToken": "TOKEN1",
    "resetTokenDate": 2025-01-16T00:00:00.000Z,
    "resetUrl": "link/TOKEN1",
  },
  {
    "_id": 2,
  },
  {
    "_id": 3,
    "resetToken": "TOKEN3",
    "resetTokenDate": 2025-01-16T00:00:00.000Z,
    "resetUrl": "link/TOKEN3",
  },
  {
    "_id": 4,
  },
]
`;
